athena-gcs/athena-gcs.yaml

```diff
@@ -0,0 +1,116 @@
+    Type: String
+    AllowedPattern: ^[a-z0-9-_]{1,64}$
+  SpillBucket:
+    Description: 'The name of the bucket where this function can spill data.'
+    Type: String
+  SpillPrefix:
+    Description: 'The prefix within SpillBucket where this function can spill data.'
+    Type: String
+    Default: athena-spill
+  LambdaTimeout:
+    Description: "Maximum Lambda invocation runtime in seconds. (min 1 - 900 max)"
+    Default: 900
+    Type: Number
+  LambdaMemory:
+    Description: "Lambda memory in MB (min 128 - 3008 max)."
+    Default: 3008
+    Type: Number
+  DisableSpillEncryption:
+    Description: "WARNING: If set to 'true' encryption for spilled data is disabled."
+    Default: "false"
+    Type: String
+  VpcId:
+    Description: 'VPC ID'
+    Type: 'AWS::EC2::VPC::Id'
+  SubnetIds:
+    Description: 'One or more Subnet IDs corresponding to the Subnet that the Lambda function can use to access you data source. (e.g. subnet1,subnet2)'
+    Type: 'List<AWS::EC2::Subnet::Id>'
+  SecurityGroupIds:
+    Description: 'One or more SecurityGroup IDs corresponding to the SecurityGroup that should be applied to the Lambda function. (e.g. sg1,sg2,sg3)'
+    Type: 'List<AWS::EC2::SecurityGroup::Id>'
+  SecretNameOrPrefix:
+    Description: 'Secret name (exact) or a prefix as a Secret Name pattern (e.g., <prefix>-<secret name>). Unless the given pattern is present in the Secret Name, invocation of GetSecretValue will be denied'
+    Type: String
+    Default: "gcs-*"
+  GcsSecretName:
+    Description: 'Secret key name in the AWS Secrets Manager.'
+    Type: String
+    Default: "gcs-athena"
+  GcsCredentialKey:
+    Description: 'GCS credential secret key that contains the GCS credentials JSON value (key in key/value).'
+    Type: String
+    Default: "gcs_credential_keys"
+  FileExtension:
+    Description: File extension to filter file names in a specific storage bucket. It can be any of PARQUET, and CSV.
+    Type: String
+  FileNamePattern:
+    Description: File name pattern when records are splitted over multiple part file. For exampel file1, file2 and so on. It does tell the appliction to collect all split recoreds into one single Table named 'file'. In this case the file pattern would be like ([a-zA-Z]+)(\d+)
+    Type: String
+Resources:
+  ConnectorConfig:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      Environment:
+        Variables:
+          disable_spill_encryption: !Ref DisableSpillEncryption
+          spill_bucket: !Ref SpillBucket
+          spill_prefix: !Ref SpillPrefix
+          gcs_secret_name: !Ref GcsSecretName
+          gcs_credential_key: !Ref GcsCredentialKey
+          file_extension: !Ref FileExtension
+          file_name_pattern: !Ref FileNamePattern
+
+      FunctionName: !Sub "${AthenaCatalogName}"
+      Handler: "com.amazonaws.athena.connectors.gcs.GcsCompositeHandler"
+      CodeUri: "./target/athena-gcs-2022.2.1.jar"

```

Comment:
This version is wrong. It should be 2022.42.2

Reply:
Changed accordingly